### PR TITLE
Create exceptions concept and finish Calculator Conundrum concept exercise

### LINKED ISSUE #2177: Calculator Conumdrum: finish implementation
settings.gradle doesn't include calculator conumdrum, it's not on the Java website either. but c# has this path available. Recommend to remove this from Java or add this to the settings.gradle and make it available to the public

### PR DESCRIPTION
**Pull request by sanderploegsma**
## New concept: `exceptions`

A new concept `exceptions` has been added, introducing the concept of exceptions in Java. It covers most of the basics like throwing exceptions, handling them with `try`/`catch`/`finally` and the difference between checked and unchecked exceptions.

## Concept exercise: Calculator Conundrum

The Calculator Conundrum exercise has been updated and IMO is now ready to be published. The changes I made:

- The `introduction.md` is rendered from a template and includes the introduction of the `exceptions` concept mentioned above.
- I made some changes to the instructions to make them a bit clearer and/or fixed some inconsistencies between the instructions and the sample code blocks
- The exception used for input validation is changed to `IllegalArgumentException`, as that more closely resembles real-world Java code and gives students the opportunity to work with multiple types of exceptions.
- I removed the `SimpleOperations` class. I know that it was probably added to let the students focus on the exception handling and not the functional behavior of the method. However, since it only wraps basic mathematical operators I think most students will not even notice it, and it may even cause confusion as to why it's there in the first place.
- Updated the tests to use AssertJ assertions to test exceptions.

Fixes #2177 

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/main/POLICIES.md#event-checklist)

### CODE REVIEW

File: concepts/exceptions/about.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+_Errors_ are exceptional conditions that are external to an application.
+An example of an error is the `OutOfMemoryError` which occurs when an application is trying to use more memory than is available on the system.
+
+Like unchecked exceptions, errors are not checked at compile-time. They are not usually thrown from application code.
+
+All exceptions in Java that inherit from `Error` are considered errors.
+
+## Throwing exceptions
+
+A method in Java can throw an exception by using the [`throw` statement][throw-statement].
```
Comment (SleeplessByte) [inline concepts/exceptions/about.md:44]: May want to explain what throwing an exception means. There are languages that don't have runtime exceptions or explicit throwing. Rust for example only has panic!, Go doesn't throw exceptions but _returns_ errors.

**Review by SleeplessByte — APPROVED** [review]
See inline

File: exercises/concept/calculator-conundrum/.docs/hints.md [hunk 2]
```diff
@@ -2,17 +2,14 @@
...
-* Throw an `IllegalOpertionException` with appropriate messages in different cases.
+## 3. Handle the exception thrown when dividing by zero
 
-## 3. Handle the thrown DivideByZero exceptions
-
-* Use a `try-catch` block to catch `ArithmeticException`
-* Pass both the `message` and the `ArithmeticException` as the `cause` parameter when throwing the exception.
\ No newline at end of file
+- Use a `try-catch` block to catch the `ArithmeticException` thrown when dividing by zero.
+- The `IllegalOperationException` class has a constructor that takes a message and a cause as its parameters.
```
[the remark this answers is not in the document]
Comment (sanderploegsma) [reply exercises/concept/calculator-conundrum/.docs/hints.md:15]: The `IllegalOperationException` is actually a custom exception in this exercise, so there is no online documentation available for it.

[the remark this answers is not in the document]
Comment (SleeplessByte) [reply exercises/concept/calculator-conundrum/.docs/hints.md:15]: Ah, TIL. It exists in some languages :P. Then ignore this comment!

**Comment by SleeplessByte** [comment]
(You can self-merge once you feel like this is addressed -- we can always make subsequent changes)

File: concepts/exceptions/.meta/config.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "blurb": "Exceptions should be thrown when an error that needs special handling occurs.",
```
Comment (ErikSchierboom) [inline concepts/exceptions/.meta/config.json:2]: Maybe make it slightly more matter-of-factly?

```suggestion
  "blurb": "Exceptions are thrown when an error that needs special handling occurs.",
```

File: config.json [hunk 4]
```diff
@@ -90,14 +90,14 @@
...
         "uuid": "9e62feec-c2c3-4fd6-94f5-0574cc65447d",
         "concepts": [
-          "exception-handling"
+          "exceptions"
         ],
         "prerequisites": [
-          "conditionals",
-          "switch-case",
-          "basics"
+          "basics",
```
Comment (ErikSchierboom) [inline config.json:96]: You can omit `basics` as this is indirectly references via the other two prerequisites (it's fine to leave it in if you want)

```suggestion
```

**Review by ErikSchierboom — APPROVED** [review]

File: concepts/exceptions/about.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
+# About
```
[the remark this answers is not in the document]
Comment (sanderploegsma) [reply concepts/exceptions/about.md:1]: Regarding multiple catch clauses: I believe I tried to cover this in the docs:

> The `try` block is followed by **one or more `catch` blocks** that catch the exceptions thrown in the `try` block.

Also the example contains two `catch` blocks to catch two different types of exceptions, and the explanation underneath shows the output of the example in both cases individually.

Do you think we need to cover it even more explicitly somehow? Any suggestions?

[the remark this answers is not in the document]
Comment (sanderploegsma) [reply concepts/exceptions/about.md:1]: Resolving this for now, if you think it should still be addressed let me know, and we'll pick it up in a new PR.

**Pull request merged**